src/compiler/checker.ts

```diff
@@ -14637,7 +14639,9 @@ namespace ts {
                     location = location.parent;
                 }
 
-                copySymbols(globals, meaning);
+                if (includeAllGlobalSymbols) {
+                    copySymbols(globals, meaning);

```

Comment:
If there is React component in global scope, we will have found it here https://github.com/Microsoft/TypeScript/pull/6006#discussion-diff-47050513R3116